yt_dlp/extractor/fifa.py

```diff
@@ -1,67 +1,244 @@
+        video_id, display_id, locale = self._match_valid_url(urlh.url).group('id', 'display_id', 'locale')
+
+        video_info = self._call_api(
+            'videoasset', video_id, 'Downloading video asset', query={'catalog': video_id})[0]
+
+        formats = []
+        subtitles = {}
+
+        for stream_type in [
+            'hls/cbcs+h265.sdr;q=0.9, hls/cbcs+h264;q=0.5, hls/clear+h264;q=0.4, mp4/;q=0.1',
+            'mpd/cenc+h264;q=0.9, mpd/clear+h264;q=0.7, mp4/;q=0.1',
+        ]:
+            session_info = self._call_api(
+                'streaming/session', video_id, 'Getting streaming session',
+                headers={'x-chili-accept-stream': stream_type},
+                data=json.dumps({'videoAssetId': video_info['id'], 'autoPlay': False}).encode())
+
+            streams_info = self._call_api(
+                'streaming/urls', video_id, 'Getting streaming urls',
+                headers={'x-chili-streaming-session': session_info['id']})
 
+            for playlist_url in traverse_obj(streams_info, (..., 'url')):
+                ext = determine_ext(playlist_url)
+                if ext == 'mpd':
+                    fmts, subs = self._extract_mpd_formats_and_subtitles(playlist_url, video_id)
+                    formats.extend(fmts)
+                    self._merge_subtitles(subs, target=subtitles)
+                elif ext == 'm3u8':
+                    fmts, subs = self._extract_m3u8_formats_and_subtitles(playlist_url, video_id, m3u8_id='hls')
+                    formats.extend(fmts)
+                    self._merge_subtitles(subs, target=subtitles)
+                else:
+                    raise ExtractorError(f'Unknown playlist URL {playlist_url}', video_id=video_id)
 
-class FifaIE(InfoExtractor):
-    _VALID_URL = r'https?://www\.fifa\.com/fifaplus/(?P<locale>\w{2})/watch/([^#?]+/)?(?P<id>\w+)'
+        self._remove_duplicate_formats(formats)
+
+        return {
+            'id': video_id,
+            'title': video_info['title'],
+            'display_id': display_id,
+            'formats': formats,
+            'subtitles': subtitles,
+            'thumbnails': [{
+                'url': update_url_query(x, {'width': 1408}),
+                'width': 1408,
+            } for x in [video_info.get('posterUrl'), video_info.get('wideCoverUrl')] if x],
+        }
+
+
+class FifaBaseIE(InfoExtractor):
+    @functools.cached_property
+    def _preconnect_link(self):
+        return self._search_regex(
+            r'<link\b[^>]+\brel\s*=\s*"preconnect"[^>]+href\s*=\s*"([^"]+)"',
+            self._download_webpage('https://fifa.com/', None), 'Preconnect Link')
+
+    def _call_api(self, path, video_id, note=None, query=None, fatal=True):
+        return self._download_json(
+            f'{self._preconnect_link}/{path}', video_id, note, query=query, fatal=fatal)
+
+
+class FifaIE(FifaBaseIE):
+    _VALID_URL = r'https?://(www\.)?fifa\.com/(fifaplus/)?(?P<locale>\w{2})/watch/(?P<id>[-\w]+)/?(?:[#?]|$)'
```

Comment:
IMO we just should not match `fifaplus` if all the content is DRM'd. Maybe add the URL pattern to `KnownDRMIE` instead

Reply:
> ... if all the content is DRM'd....

Let me check. IIRC some videos in FIFA Article are not DRM. I think I may have to delete a bunch of code lines. :(